File: python/7 kyu/381.py
# https://www.codewars.com/kata/5b2117eea454c89d4400005f/train/python

# Passed

def leonardo_numbers(n, L0, L1, add) :
    result = []
    for iter_num in range(1, n + 1):
        if iter_num == 1:
            result.append(L0)
        elif iter_num == 2:
            result.append(L1)
        else:
            val = result[-1] + result[-2] + add
            result.append(val)
            
    return result

output = leonardo_numbers(5, 1, 1, 1)
print(output)

File: nodejs/7 kyu/381.js

// https://www.codewars.com/kata/5acc79efc6fde7838a0000a0/train/javascript

// Passed

class Node {
  constructor(value, left = null, right = null) {
    this.value = value;
    this.left = left;
    this.right = right;
  }
}

function search(n, root) {
  if (root === null) {
    return false;
  } else if (root.value === n) {
    return true;
  }

  return search(n, root.left) || search(n, root.right);
}

const output = search(555, new Node(666, new Node(555), new Node(444)));
console.log(output);